File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabble

import "strings"

func Score(word string) int {

	var scrabbleScore = make(map[string]int)
	scrabbleScore["A"] = 1
	scrabbleScore["E"] = 1
	scrabbleScore["I"] = 1
	scrabbleScore["O"] = 1
	scrabbleScore["U"] = 1
	scrabbleScore["L"] = 1
	scrabbleScore["N"] = 1
	scrabbleScore["R"] = 1
	scrabbleScore["S"] = 1
	scrabbleScore["T"] = 1

	scrabbleScore["D"] = 2
	scrabbleScore["G"] = 2

	scrabbleScore["B"] = 3
	scrabbleScore["C"] = 3
	scrabbleScore["M"] = 3
	scrabbleScore["P"] = 3

	scrabbleScore["F"] = 4
	scrabbleScore["H"] = 4
	scrabbleScore["V"] = 4
	scrabbleScore["W"] = 4
	scrabbleScore["Y"] = 4

	scrabbleScore["K"] = 5

	scrabbleScore["J"] = 8
	scrabbleScore["X"] = 8

	scrabbleScore["Q"] = 10
	scrabbleScore["Z"] = 10

	var count = 0

	for _, c := range word {
		count += scrabbleScore[strings.ToUpper(string(c))]
	}

	return count

}

File: solutions/c/scrabble-score/1/scrabble_score.c
#include "scrabble_score.h"

#include <stdint.h>
#include <stdlib.h>


unsigned int score(const char *word) {
    static const int letter_scores[26] = {1, 3, 3, 2, 1, 4, 2, 4, 1, 8, 5, 1, 3, 1, 1, 3, 10, 1, 1, 1, 1, 4, 4, 8, 4, 10};
    unsigned int total_score = 0;
    for (size_t i = 0; word[i] != '\0'; i++) {
        char c = word[i];
        if (c >= 'a' && c <= 'z') {
            total_score += letter_scores[c - 'a'];
        } else if (c >= 'A' && c <= 'Z') {
            total_score += letter_scores[c - 'A'];
        }
    }
    return total_score;
}
